.changeset/five-jars-hear.md

```diff
@@ -0,0 +1,9 @@
+---
+'astro': patch
+---
+
+Removes the `experimental.directRenderScript` option and always render script directly. This new strategy prevents scripts from being executed in pages where they are not used. 
+
+Scripts will directly render as declared in Astro files (including existing features like TypeScript, importing `node_modules`, and deduplicating scripts). You can also now conditionally render scripts in your Astro file.
+
+However, this means scripts are no longer hoisted to the `<head>`, multiple scripts on a page are no longer bundled together, and the `<script>` tag may interfere with the CSS styling. Make sure to review your script tags and ensure that they behave as expected.
```

Comment:
```suggestion
Updates Astro's default `<script>` execution strategy and removes the `experimental.directRenderScript` option as this is now the default behavior: scripts are always rendered directly. This new strategy prevents scripts from being executed in pages where they are not used. 

Scripts will directly render as declared in Astro files (including existing features like TypeScript, importing `node_modules`, and deduplicating scripts). You can also now conditionally render scripts in your Astro file.

However, this means scripts are no longer hoisted to the `<head>`, multiple scripts on a page are no longer bundled together, and the `<script>` tag may interfere with the CSS styling.

As this is a potentially breaking change to your script behavior, please review your `<script>` tags and ensure that they behave as expected.
```
This was great, @bluwy ! I swapped the order in the first line to emphasize the change to default behavior because that affects more people, vs people who might think, "Oh, I can skip this because I never used that experimental flag."  :smile:

Reply:
Good call. I think there's one minor thing I'll update, which is "`<script>` rendering strategy" instead. Otherwise looks great!